packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -1,7 +1,96 @@
+import "@paprika/helpers/lib/polyfills/elementClosest";
 import useListBox from "../../../useListBox";
 import invokeOnChange from "../../../helpers/invokeOnChange";
 import { KEY_PRESS, CLICK } from "../../../types";
 
+function getOptions(event) {
+  let list = null;
+  if (event.target.dataset.pkaAnchor && event.target.dataset.pkaAnchor === "list-filter-input") {
+    // the event is coming from the filter
+    list = event.target.closest("[data-pka-anchor='list-box-box']").querySelector("[data-pka-anchor='styled-list']");
+  } else {
+    list = event.target.querySelector("[data-pka-anchor='styled-list']");
+  }
+
+  // filtering only results with role="option" attribute
+  return [...list.children].filter(li => li.hasAttribute("role"));
+}
+
+function getNextUp(event) {
+  function next(element) {
+    const list = event.target.closest("[data-pka-anchor='styled-list']");
+    const options = list.children;
+    if (list && [...options].indexOf(element) > 0) {
```

Comment:
same here, `[...options]` is redundant. 

We don't need to create a shallow copy of the array or object, unless 

1. we actually want to modify this array\object
2. we actually care about immutability in the specific context

For instance : 

```js
function getMaxNumber(numbers) {
   const n = [...numbers];
   
   let max = n[0]; 
   
   for (i = 1; i < n.length; i++) {
     max = max > n[i] : max : n[i]
   }
   
   return max;
}

getMaxNumber([4,6,12,76,3])
```
☝️ here we don't care if the input array mutates, and we don't even mutate it - we only iterate through it. Therefore, no need to do `const n = [...numbers];`

Reply:
Also, don't forget that even if we do `[...arr]` it is a shallow copy. So, 

```js
const oldArray = [{ name: 'Bob' }];
const newArray = [...oldArray];

newArray[0].name = 'John';
console.log(oldArray[0].name) //  'John'
```

Reply:
This is a node list, not an array, I'm converting it into an array to be able to use the methods on it unless you mean something different?